lib/membrane_wav/serializer.ex

```diff
@@ -77,20 +95,33 @@ defmodule Membrane.WAV.Serializer do
         buffers_count,
         :buffers,
         context,
-        %{header_created: true, frames_per_buffer: frames} = state
+        %{frames_per_buffer: frames} = state
       ) do
     caps = context.pads.output.caps
     demand_size = Caps.frames_to_bytes(frames, caps) * buffers_count
     {{:ok, demand: {:input, demand_size}}, state}
   end
 
   @impl true
-  def handle_process(:input, buffer, _context, %{header_created: true} = state) do
-    {{:ok, buffer: {:output, buffer}, redemand: :output}, state}
+  def handle_process_list(:input, _buffers, _context, %{header_length: 0}) do
+    raise(RuntimeError, "buffers received before caps, so the header is not created yet")
```

Comment:
If not using any custom exception, use `raise/1`
```suggestion
    raise "Buffers received before caps, cannot create the header"
```